Can't run configure.sh

I'm just trying to check this out, but I can't successfully run `configure.sh`:

```
    LINK redis-server
clang: error: no such file or directory: '../deps/hiredis/libhiredis.a'
clang: error: no such file or directory: '../deps/lua/src/liblua.a'
```

```
$ ./configure.sh
~/src/redis-roaring ~/src/redis-roaring
/Users/abernstein/src/redis-roaring/deps/CRoaring
We are about to amalgamate all CRoaring files into one source file.
See https://www.sqlite.org/amalgamation.html and https://en.wikipedia.org/wiki/Single_Compilation_Unit for rationale.
Creating roaring.h...
Creating roaring.c...
Creating amalgamation_demo.c...
Done with C amalgamation. Proceeding with C++ wrap.
Creating roaring.hh...
Creating amalgamation_demo.cpp...
Done with C++.
Done with all files generation.
Files have been written to current directory: /Users/abernstein/src/redis-roaring/deps/CRoaring
-rw-r--r-- 1 abernstein staff    354 Jun 30 13:31 amalgamation_demo.c
-rw-r--r-- 1 abernstein staff    486 Jun 30 13:31 amalgamation_demo.cpp
-rw-r--r-- 1 abernstein staff 415392 Jun 30 13:31 roaring.c
-rw-r--r-- 1 abernstein staff 256231 Jun 30 13:31 roaring.h
-rw-r--r-- 1 abernstein staff  51834 Jun 30 13:31 roaring.hh
Giving final instructions:

The interface is found in the file 'include/roaring/roaring.h'.

Try :
cc -march=native -O3 -std=c11  -o amalgamation_demo amalgamation_demo.c  && ./amalgamation_demo

For C++, try :
c++ -march=native -O3 -std=c++11 -o amalgamation_demo amalgamation_demo.cpp  && ./amalgamation_demo

You can build a shared library with the following command :
cc -march=native -O3 -std=c11 -shared -o libroaring.dylib -fPIC roaring.c
You can try compiling with the extra flags -DDISABLEAVX or -DDISABLE_X64 to disable AVX and all x64 optimization respectively.
/Users/abernstein/src/redis-roaring/deps/CRoaring/build
-- BENCHMARK_DATA_DIR: /Users/abernstein/src/redis-roaring/deps/CRoaring/benchmarks/realdata/
-- TEST_DATA_DIR: /Users/abernstein/src/redis-roaring/deps/CRoaring/tests/testdata/
-- Building a static library.
-- ROARING_LIB_TYPE: STATIC
-- Library output directory (does not apply to Visual Studio): /Users/abernstein/src/redis-roaring/deps/CRoaring/build
-- CMAKE_SYSTEM_PROCESSOR: x86_64
-- CMAKE_BUILD_TYPE: Release
-- DISABLE_X64: OFF
-- DISABLE_AVX: OFF
-- BUILD_STATIC: ON
-- BUILD_LTO: OFF
-- SANITIZE: OFF
-- CMAKE_C_COMPILER: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- CMAKE_C_FLAGS: -std=c11 -fPIC -march=native  -Wall -Wextra -Wsign-compare -Wshadow -Wwrite-strings -Wpointer-arith -Winit-self
-- CMAKE_C_FLAGS_DEBUG: -ggdb
-- CMAKE_C_FLAGS_RELEASE: -O3
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/abernstein/src/redis-roaring/deps/CRoaring/build
[ 34%] Built target roaring
[ 38%] Built target array_container_benchmark
[ 42%] Built target run_container_benchmark
[ 46%] Built target bitset_container_benchmark
[ 50%] Built target real_bitmaps_benchmark
[ 53%] Built target real_bitmaps_contains_benchmark
[ 57%] Built target create_benchmark
[ 61%] Built target cmocka
[ 65%] Built target container_comparison_unit
[ 69%] Built target format_portability_unit
[ 73%] Built target array_container_unit
[ 76%] Built target realdata_unit
[ 80%] Built target toplevel_unit
[ 84%] Built target run_container_unit
[ 88%] Built target mixed_container_unit
[ 92%] Built target util_unit
[ 96%] Built target bitset_container_unit
[100%] Built target cpp_unit
~/src/redis-roaring
/Users/abernstein/src/redis-roaring/deps/redis
cd src && /Applications/Xcode.app/Contents/Developer/usr/bin/make all
/Users/abernstein/src/redis-roaring/deps/redis/src
    LINK redis-server
clang: error: no such file or directory: '../deps/hiredis/libhiredis.a'
clang: error: no such file or directory: '../deps/lua/src/liblua.a'
make[1]: *** [redis-server] Error 1
make: *** [all] Error 2
/Users/abernstein/src/redis-roaring
/Users/abernstein/src/redis-roaring/deps/hiredis
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb x86_64 net.c
clang: error: no such file or directory: 'x86_64'
make: *** [net.o] Error 1
/Users/abernstein/src/redis-roaring
```

Thanks @lemire. After following your exact instructions I see the below output from `./configure.sh`. The `git submodule` commands appear to complete without issue.

I wonder if there is something screwy on my system, I may just try on a different machine.

```
$ ./configure.sh
~/src/redis-roaring ~/src/redis-roaring
/Users/abernstein/src/redis-roaring/deps/CRoaring
We are about to amalgamate all CRoaring files into one source file.
See https://www.sqlite.org/amalgamation.html and https://en.wikipedia.org/wiki/Single_Compilation_Unit for rationale.
Creating roaring.h...
Creating roaring.c...
Creating amalgamation_demo.c...
Done with C amalgamation. Proceeding with C++ wrap.
Creating roaring.hh...
Creating amalgamation_demo.cpp...
Done with C++.
Done with all files generation.
Files have been written to current directory: /Users/abernstein/src/redis-roaring/deps/CRoaring
-rw-r--r-- 1 abernstein staff    354 Jul 21 13:27 amalgamation_demo.c
-rw-r--r-- 1 abernstein staff    486 Jul 21 13:27 amalgamation_demo.cpp
-rw-r--r-- 1 abernstein staff 415392 Jul 21 13:27 roaring.c
-rw-r--r-- 1 abernstein staff 256231 Jul 21 13:27 roaring.h
-rw-r--r-- 1 abernstein staff  51834 Jul 21 13:27 roaring.hh
Giving final instructions:

The interface is found in the file 'include/roaring/roaring.h'.

Try :
cc -march=native -O3 -std=c11  -o amalgamation_demo amalgamation_demo.c  && ./amalgamation_demo

For C++, try :
c++ -march=native -O3 -std=c++11 -o amalgamation_demo amalgamation_demo.cpp  && ./amalgamation_demo

You can build a shared library with the following command :
cc -march=native -O3 -std=c11 -shared -o libroaring.dylib -fPIC roaring.c
You can try compiling with the extra flags -DDISABLEAVX or -DDISABLE_X64 to disable AVX and all x64 optimization respectively.
/Users/abernstein/src/redis-roaring/deps/CRoaring/build
-- No build type selected, default to Release
-- The C compiler identification is AppleClang 8.1.0.8020042
-- The CXX compiler identification is AppleClang 8.1.0.8020042
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- BENCHMARK_DATA_DIR: /Users/abernstein/src/redis-roaring/deps/CRoaring/benchmarks/realdata/
-- TEST_DATA_DIR: /Users/abernstein/src/redis-roaring/deps/CRoaring/tests/testdata/
-- Building a static library.
-- ROARING_LIB_TYPE: STATIC
-- Library output directory (does not apply to Visual Studio): /Users/abernstein/src/redis-roaring/deps/CRoaring/build
-- CMAKE_SYSTEM_PROCESSOR: x86_64
-- CMAKE_BUILD_TYPE: Release
-- DISABLE_X64: OFF
-- DISABLE_AVX: OFF
-- BUILD_STATIC: ON
-- BUILD_LTO: OFF
-- SANITIZE: OFF
-- CMAKE_C_COMPILER: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- CMAKE_C_FLAGS: -std=c11 -fPIC -march=native  -Wall -Wextra -Wsign-compare -Wshadow -Wwrite-strings -Wpointer-arith -Winit-self
-- CMAKE_C_FLAGS_DEBUG: -ggdb
-- CMAKE_C_FLAGS_RELEASE: -O3
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/abernstein/src/redis-roaring/deps/CRoaring/build
Scanning dependencies of target roaring
[  1%] Building C object src/CMakeFiles/roaring.dir/array_util.c.o
[  3%] Building C object src/CMakeFiles/roaring.dir/bitset_util.c.o
[  5%] Building C object src/CMakeFiles/roaring.dir/containers/array.c.o
[  7%] Building C object src/CMakeFiles/roaring.dir/containers/bitset.c.o
[  9%] Building C object src/CMakeFiles/roaring.dir/containers/containers.c.o
[ 11%] Building C object src/CMakeFiles/roaring.dir/containers/convert.c.o
[ 13%] Building C object src/CMakeFiles/roaring.dir/containers/mixed_intersection.c.o
[ 15%] Building C object src/CMakeFiles/roaring.dir/containers/mixed_union.c.o
[ 17%] Building C object src/CMakeFiles/roaring.dir/containers/mixed_equal.c.o
[ 19%] Building C object src/CMakeFiles/roaring.dir/containers/mixed_subset.c.o
[ 21%] Building C object src/CMakeFiles/roaring.dir/containers/mixed_negation.c.o
[ 23%] Building C object src/CMakeFiles/roaring.dir/containers/mixed_xor.c.o
[ 25%] Building C object src/CMakeFiles/roaring.dir/containers/mixed_andnot.c.o
[ 26%] Building C object src/CMakeFiles/roaring.dir/containers/run.c.o
[ 28%] Building C object src/CMakeFiles/roaring.dir/roaring.c.o
[ 30%] Building C object src/CMakeFiles/roaring.dir/roaring_priority_queue.c.o
[ 32%] Building C object src/CMakeFiles/roaring.dir/roaring_array.c.o
[ 34%] Linking C static library libroaring.a
[ 34%] Built target roaring
Scanning dependencies of target array_container_benchmark
[ 36%] Building C object benchmarks/CMakeFiles/array_container_benchmark.dir/array_container_benchmark.c.o
[ 38%] Linking C executable array_container_benchmark
[ 38%] Built target array_container_benchmark
Scanning dependencies of target run_container_benchmark
[ 40%] Building C object benchmarks/CMakeFiles/run_container_benchmark.dir/run_container_benchmark.c.o
[ 42%] Linking C executable run_container_benchmark
[ 42%] Built target run_container_benchmark
Scanning dependencies of target bitset_container_benchmark
[ 44%] Building C object benchmarks/CMakeFiles/bitset_container_benchmark.dir/bitset_container_benchmark.c.o
[ 46%] Linking C executable bitset_container_benchmark
[ 46%] Built target bitset_container_benchmark
Scanning dependencies of target real_bitmaps_benchmark
[ 48%] Building C object benchmarks/CMakeFiles/real_bitmaps_benchmark.dir/real_bitmaps_benchmark.c.o
[ 50%] Linking C executable real_bitmaps_benchmark
[ 50%] Built target real_bitmaps_benchmark
Scanning dependencies of target real_bitmaps_contains_benchmark
[ 51%] Building C object benchmarks/CMakeFiles/real_bitmaps_contains_benchmark.dir/real_bitmaps_contains_benchmark.c.o
[ 53%] Linking C executable real_bitmaps_contains_benchmark
[ 53%] Built target real_bitmaps_contains_benchmark
Scanning dependencies of target create_benchmark
[ 55%] Building C object benchmarks/CMakeFiles/create_benchmark.dir/create_benchmark.c.o
[ 57%] Linking C executable create_benchmark
[ 57%] Built target create_benchmark
Scanning dependencies of target cmocka
[ 59%] Building C object tests/vendor/cmocka/CMakeFiles/cmocka.dir/cmocka.c.o
[ 61%] Linking C static library libcmocka.a
[ 61%] Built target cmocka
Scanning dependencies of target container_comparison_unit
[ 63%] Building C object tests/CMakeFiles/container_comparison_unit.dir/container_comparison_unit.c.o
[ 65%] Linking C executable container_comparison_unit
[ 65%] Built target container_comparison_unit
Scanning dependencies of target format_portability_unit
[ 67%] Building C object tests/CMakeFiles/format_portability_unit.dir/format_portability_unit.c.o
[ 69%] Linking C executable format_portability_unit
[ 69%] Built target format_portability_unit
Scanning dependencies of target array_container_unit
[ 71%] Building C object tests/CMakeFiles/array_container_unit.dir/array_container_unit.c.o
[ 73%] Linking C executable array_container_unit
[ 73%] Built target array_container_unit
Scanning dependencies of target realdata_unit
[ 75%] Building C object tests/CMakeFiles/realdata_unit.dir/realdata_unit.c.o
[ 76%] Linking C executable realdata_unit
[ 76%] Built target realdata_unit
Scanning dependencies of target toplevel_unit
[ 78%] Building C object tests/CMakeFiles/toplevel_unit.dir/toplevel_unit.c.o
[ 80%] Linking C executable toplevel_unit
[ 80%] Built target toplevel_unit
Scanning dependencies of target run_container_unit
[ 82%] Building C object tests/CMakeFiles/run_container_unit.dir/run_container_unit.c.o
[ 84%] Linking C executable run_container_unit
[ 84%] Built target run_container_unit
Scanning dependencies of target mixed_container_unit
[ 86%] Building C object tests/CMakeFiles/mixed_container_unit.dir/mixed_container_unit.c.o
[ 88%] Linking C executable mixed_container_unit
[ 88%] Built target mixed_container_unit
Scanning dependencies of target util_unit
[ 90%] Building C object tests/CMakeFiles/util_unit.dir/util_unit.c.o
[ 92%] Linking C executable util_unit
[ 92%] Built target util_unit
Scanning dependencies of target bitset_container_unit
[ 94%] Building C object tests/CMakeFiles/bitset_container_unit.dir/bitset_container_unit.c.o
[ 96%] Linking C executable bitset_container_unit
[ 96%] Built target bitset_container_unit
Scanning dependencies of target cpp_unit
[ 98%] Building CXX object tests/CMakeFiles/cpp_unit.dir/cpp_unit.cpp.o
[100%] Linking CXX executable cpp_unit
[100%] Built target cpp_unit
~/src/redis-roaring
/Users/abernstein/src/redis-roaring/deps/redis
cd src && /Applications/Xcode.app/Contents/Developer/usr/bin/make all
/Users/abernstein/src/redis-roaring/deps/redis/src
    CC Makefile.dep
rm -rf redis-server redis-sentinel redis-cli redis-benchmark redis-check-rdb redis-check-aof *.o *.gcda *.gcno *.gcov redis.info lcov-html Makefile.dep dict-benchmark
(cd ../deps && /Applications/Xcode.app/Contents/Developer/usr/bin/make distclean)
(cd hiredis && /Applications/Xcode.app/Contents/Developer/usr/bin/make clean) > /dev/null || true
(cd linenoise && /Applications/Xcode.app/Contents/Developer/usr/bin/make clean) > /dev/null || true
(cd lua && /Applications/Xcode.app/Contents/Developer/usr/bin/make clean) > /dev/null || true
(cd jemalloc && [ -f Makefile ] && /Applications/Xcode.app/Contents/Developer/usr/bin/make distclean) > /dev/null || true
(rm -f .make-*)
(rm -f .make-*)
echo STD=-std=c99 -pedantic -DREDIS_STATIC='' >> .make-settings
echo WARN=-Wall -W -Wno-missing-field-initializers >> .make-settings
echo OPT=-O2 >> .make-settings
echo MALLOC=libc >> .make-settings
echo CFLAGS= >> .make-settings
echo LDFLAGS= >> .make-settings
echo REDIS_CFLAGS= >> .make-settings
echo REDIS_LDFLAGS= >> .make-settings
echo PREV_FINAL_CFLAGS=-std=c99 -pedantic -DREDIS_STATIC='' -Wall -W -Wno-missing-field-initializers -O2 -g -ggdb   -I../deps/hiredis -I../deps/linenoise -I../deps/lua/src >> .make-settings
echo PREV_FINAL_LDFLAGS=  -g -ggdb >> .make-settings
(cd ../deps && /Applications/Xcode.app/Contents/Developer/usr/bin/make hiredis linenoise lua)
(cd hiredis && /Applications/Xcode.app/Contents/Developer/usr/bin/make clean) > /dev/null || true
(cd linenoise && /Applications/Xcode.app/Contents/Developer/usr/bin/make clean) > /dev/null || true
(cd lua && /Applications/Xcode.app/Contents/Developer/usr/bin/make clean) > /dev/null || true
(cd jemalloc && [ -f Makefile ] && /Applications/Xcode.app/Contents/Developer/usr/bin/make distclean) > /dev/null || true
(rm -f .make-*)
(echo "" > .make-ldflags)
(echo "" > .make-cflags)
MAKE hiredis
cd hiredis && /Applications/Xcode.app/Contents/Developer/usr/bin/make static
/Users/abernstein/src/redis-roaring/deps/redis/deps/hiredis
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb x86_64 net.c
clang: error: no such file or directory: 'x86_64'
make[3]: *** [net.o] Error 1
make[2]: *** [hiredis] Error 2
make[1]: [persist-settings] Error 2 (ignored)
    CC adlist.o
    CC quicklist.o
    CC ae.o
    CC anet.o
    CC dict.o
    CC server.o
    CC sds.o
    CC zmalloc.o
    CC lzf_c.o
    CC lzf_d.o
    CC pqsort.o
    CC zipmap.o
    CC sha1.o
    CC ziplist.o
    CC release.o
    CC networking.o
    CC util.o
    CC object.o
    CC db.o
    CC replication.o
    CC rdb.o
    CC t_string.o
    CC t_list.o
    CC t_set.o
    CC t_zset.o
    CC t_hash.o
    CC config.o
    CC aof.o
    CC pubsub.o
    CC multi.o
    CC debug.o
    CC sort.o
    CC intset.o
    CC syncio.o
    CC cluster.o
    CC crc16.o
    CC endianconv.o
    CC slowlog.o
    CC scripting.o
    CC bio.o
    CC rio.o
    CC rand.o
    CC memtest.o
    CC crc64.o
    CC bitops.o
    CC sentinel.o
    CC notify.o
    CC setproctitle.o
    CC blocked.o
    CC hyperloglog.o
    CC latency.o
    CC sparkline.o
    CC redis-check-rdb.o
    CC geo.o
    CC lazyfree.o
    CC module.o
    CC evict.o
    CC expire.o
    CC geohash.o
    CC geohash_helper.o
    CC childinfo.o
    CC defrag.o
    CC siphash.o
    CC rax.o
    LINK redis-server
clang: error: no such file or directory: '../deps/hiredis/libhiredis.a'
clang: error: no such file or directory: '../deps/lua/src/liblua.a'
make[1]: *** [redis-server] Error 1
make: *** [all] Error 2
/Users/abernstein/src/redis-roaring
/Users/abernstein/src/redis-roaring/deps/hiredis
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb x86_64 net.c
clang: error: no such file or directory: 'x86_64'
make: *** [net.o] Error 1
/Users/abernstein/src/redis-roaring
```

I find this error ``clang: error: no such file or directory: 'x86_64'`` suspicious. 



Can you do...

```
$ cc --version
```

Then 

```
$ cd  deps/hiredis
$ make clean
$ make
```

Here is what you should see:


```
$ make
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb  net.c
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb  hiredis.c
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb  sds.c
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb  async.c
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb  read.c
cc -shared -Wl,-install_name,libhiredis.0.13.dylib -o libhiredis.dylib  net.o hiredis.o sds.o async.o read.o
ar rcs libhiredis.a net.o hiredis.o sds.o async.o read.o
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb  test.c
cc -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb  -o hiredis-test   test.o libhiredis.a
Generating hiredis.pc for pkgconfig...
```


If that does not happen, create a small file called hello.c in the same directory...

```
#include <stdio.h>
int main(void){
  printf("hello\n");
  return 0;
}
```

and type...

```
$ cc -o hello hello.c
$ ./hello
```



Here is what I get in full (on my Mac):

```
$ ./configure.sh
~/CVS/github/redis-roaring ~/CVS/github/redis-roaring
We are about to amalgamate all CRoaring files into one source file.
See https://www.sqlite.org/amalgamation.html and https://en.wikipedia.org/wiki/Single_Compilation_Unit for rationale.
Creating roaring.h...
Creating roaring.c...
Creating amalgamation_demo.c...
Done with C amalgamation. Proceeding with C++ wrap.
Creating roaring.hh...
Creating amalgamation_demo.cpp...
Done with C++.
Done with all files generation.
Files have been written to current directory: /Users/lemire/CVS/github/redis-roaring/deps/CRoaring
-rw-r--r--  1 lemire  staff     354 21 Jul 14:16 amalgamation_demo.c
-rw-r--r--  1 lemire  staff     486 21 Jul 14:16 amalgamation_demo.cpp
-rw-r--r--  1 lemire  staff  415494 21 Jul 14:16 roaring.c
-rw-r--r--  1 lemire  staff  256363 21 Jul 14:16 roaring.h
-rw-r--r--  1 lemire  staff   51846 21 Jul 14:16 roaring.hh
Giving final instructions:

The interface is found in the file 'include/roaring/roaring.h'.

Try :
cc -march=native -O3 -std=c11  -o amalgamation_demo amalgamation_demo.c  && ./amalgamation_demo

For C++, try :
c++ -march=native -O3 -std=c++11 -o amalgamation_demo amalgamation_demo.cpp  && ./amalgamation_demo

You can build a shared library with the following command :
cc -march=native -O3 -std=c11 -shared -o libroaring.dylib -fPIC roaring.c
You can try compiling with the extra flags -DDISABLEAVX or -DDISABLE_X64 to disable AVX and all x64 optimization respectively.
-- No build type selected, default to Release
-- The C compiler identification is AppleClang 8.1.0.8020042
-- The CXX compiler identification is AppleClang 8.1.0.8020042
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- BENCHMARK_DATA_DIR: /Users/lemire/CVS/github/redis-roaring/deps/CRoaring/benchmarks/realdata/
-- TEST_DATA_DIR: /Users/lemire/CVS/github/redis-roaring/deps/CRoaring/tests/testdata/
-- Building a static library.
-- ROARING_LIB_TYPE: STATIC
-- Library output directory (does not apply to Visual Studio): /Users/lemire/CVS/github/redis-roaring/deps/CRoaring/build
-- CMAKE_SYSTEM_PROCESSOR: x86_64
-- CMAKE_BUILD_TYPE: Release
-- DISABLE_X64: OFF
-- DISABLE_AVX: OFF
-- BUILD_STATIC: ON
-- BUILD_LTO: OFF
-- SANITIZE: OFF
-- CMAKE_C_COMPILER: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- CMAKE_C_FLAGS: -std=c11 -fPIC -march=native  -Wall -Wextra -Wsign-compare -Wshadow -Wwrite-strings -Wpointer-arith -Winit-self
-- CMAKE_C_FLAGS_DEBUG: -ggdb
-- CMAKE_C_FLAGS_RELEASE: -O3
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/lemire/CVS/github/redis-roaring/deps/CRoaring/build
Scanning dependencies of target roaring
[  1%] Building C object src/CMakeFiles/roaring.dir/array_util.c.o
[  3%] Building C object src/CMakeFiles/roaring.dir/bitset_util.c.o
[  5%] Building C object src/CMakeFiles/roaring.dir/containers/array.c.o
[  7%] Building C object src/CMakeFiles/roaring.dir/containers/bitset.c.o
[  9%] Building C object src/CMakeFiles/roaring.dir/containers/containers.c.o
[ 11%] Building C object src/CMakeFiles/roaring.dir/containers/convert.c.o
[ 13%] Building C object src/CMakeFiles/roaring.dir/containers/mixed_intersection.c.o
[ 15%] Building C object src/CMakeFiles/roaring.dir/containers/mixed_union.c.o
[ 17%] Building C object src/CMakeFiles/roaring.dir/containers/mixed_equal.c.o
[ 19%] Building C object src/CMakeFiles/roaring.dir/containers/mixed_subset.c.o
[ 21%] Building C object src/CMakeFiles/roaring.dir/containers/mixed_negation.c.o
[ 23%] Building C object src/CMakeFiles/roaring.dir/containers/mixed_xor.c.o
[ 25%] Building C object src/CMakeFiles/roaring.dir/containers/mixed_andnot.c.o
[ 26%] Building C object src/CMakeFiles/roaring.dir/containers/run.c.o
[ 28%] Building C object src/CMakeFiles/roaring.dir/roaring.c.o
[ 30%] Building C object src/CMakeFiles/roaring.dir/roaring_priority_queue.c.o
[ 32%] Building C object src/CMakeFiles/roaring.dir/roaring_array.c.o
[ 34%] Linking C static library libroaring.a
[ 34%] Built target roaring
Scanning dependencies of target array_container_benchmark
[ 36%] Building C object benchmarks/CMakeFiles/array_container_benchmark.dir/array_container_benchmark.c.o
[ 38%] Linking C executable array_container_benchmark
[ 38%] Built target array_container_benchmark
Scanning dependencies of target run_container_benchmark
[ 40%] Building C object benchmarks/CMakeFiles/run_container_benchmark.dir/run_container_benchmark.c.o
[ 42%] Linking C executable run_container_benchmark
[ 42%] Built target run_container_benchmark
Scanning dependencies of target bitset_container_benchmark
[ 44%] Building C object benchmarks/CMakeFiles/bitset_container_benchmark.dir/bitset_container_benchmark.c.o
[ 46%] Linking C executable bitset_container_benchmark
[ 46%] Built target bitset_container_benchmark
Scanning dependencies of target real_bitmaps_benchmark
[ 48%] Building C object benchmarks/CMakeFiles/real_bitmaps_benchmark.dir/real_bitmaps_benchmark.c.o
[ 50%] Linking C executable real_bitmaps_benchmark
[ 50%] Built target real_bitmaps_benchmark
Scanning dependencies of target real_bitmaps_contains_benchmark
[ 51%] Building C object benchmarks/CMakeFiles/real_bitmaps_contains_benchmark.dir/real_bitmaps_contains_benchmark.c.o
[ 53%] Linking C executable real_bitmaps_contains_benchmark
[ 53%] Built target real_bitmaps_contains_benchmark
Scanning dependencies of target create_benchmark
[ 55%] Building C object benchmarks/CMakeFiles/create_benchmark.dir/create_benchmark.c.o
[ 57%] Linking C executable create_benchmark
[ 57%] Built target create_benchmark
Scanning dependencies of target cmocka
[ 59%] Building C object tests/vendor/cmocka/CMakeFiles/cmocka.dir/cmocka.c.o
[ 61%] Linking C static library libcmocka.a
[ 61%] Built target cmocka
Scanning dependencies of target container_comparison_unit
[ 63%] Building C object tests/CMakeFiles/container_comparison_unit.dir/container_comparison_unit.c.o
[ 65%] Linking C executable container_comparison_unit
[ 65%] Built target container_comparison_unit
Scanning dependencies of target cpp_unit
[ 67%] Building CXX object tests/CMakeFiles/cpp_unit.dir/cpp_unit.cpp.o
[ 69%] Linking CXX executable cpp_unit
[ 69%] Built target cpp_unit
Scanning dependencies of target format_portability_unit
[ 71%] Building C object tests/CMakeFiles/format_portability_unit.dir/format_portability_unit.c.o
[ 73%] Linking C executable format_portability_unit
[ 73%] Built target format_portability_unit
Scanning dependencies of target array_container_unit
[ 75%] Building C object tests/CMakeFiles/array_container_unit.dir/array_container_unit.c.o
[ 76%] Linking C executable array_container_unit
[ 76%] Built target array_container_unit
Scanning dependencies of target realdata_unit
[ 78%] Building C object tests/CMakeFiles/realdata_unit.dir/realdata_unit.c.o
[ 80%] Linking C executable realdata_unit
[ 80%] Built target realdata_unit
Scanning dependencies of target toplevel_unit
[ 82%] Building C object tests/CMakeFiles/toplevel_unit.dir/toplevel_unit.c.o
[ 84%] Linking C executable toplevel_unit
[ 84%] Built target toplevel_unit
Scanning dependencies of target run_container_unit
[ 86%] Building C object tests/CMakeFiles/run_container_unit.dir/run_container_unit.c.o
[ 88%] Linking C executable run_container_unit
[ 88%] Built target run_container_unit
Scanning dependencies of target mixed_container_unit
[ 90%] Building C object tests/CMakeFiles/mixed_container_unit.dir/mixed_container_unit.c.o
[ 92%] Linking C executable mixed_container_unit
[ 92%] Built target mixed_container_unit
Scanning dependencies of target util_unit
[ 94%] Building C object tests/CMakeFiles/util_unit.dir/util_unit.c.o
[ 96%] Linking C executable util_unit
[ 96%] Built target util_unit
Scanning dependencies of target bitset_container_unit
[ 98%] Building C object tests/CMakeFiles/bitset_container_unit.dir/bitset_container_unit.c.o
[100%] Linking C executable bitset_container_unit
[100%] Built target bitset_container_unit
~/CVS/github/redis-roaring
cd src && /Applications/Xcode.app/Contents/Developer/usr/bin/make all
    CC Makefile.dep
rm -rf redis-server redis-sentinel redis-cli redis-benchmark redis-check-rdb redis-check-aof *.o *.gcda *.gcno *.gcov redis.info lcov-html Makefile.dep dict-benchmark
(cd ../deps && /Applications/Xcode.app/Contents/Developer/usr/bin/make distclean)
(cd hiredis && /Applications/Xcode.app/Contents/Developer/usr/bin/make clean) > /dev/null || true
(cd linenoise && /Applications/Xcode.app/Contents/Developer/usr/bin/make clean) > /dev/null || true
(cd lua && /Applications/Xcode.app/Contents/Developer/usr/bin/make clean) > /dev/null || true
(cd jemalloc && [ -f Makefile ] && /Applications/Xcode.app/Contents/Developer/usr/bin/make distclean) > /dev/null || true
(rm -f .make-*)
(rm -f .make-*)
echo STD=-std=c99 -pedantic -DREDIS_STATIC='' >> .make-settings
echo WARN=-Wall -W -Wno-missing-field-initializers >> .make-settings
echo OPT=-O2 >> .make-settings
echo MALLOC=libc >> .make-settings
echo CFLAGS= >> .make-settings
echo LDFLAGS= >> .make-settings
echo REDIS_CFLAGS= >> .make-settings
echo REDIS_LDFLAGS= >> .make-settings
echo PREV_FINAL_CFLAGS=-std=c99 -pedantic -DREDIS_STATIC='' -Wall -W -Wno-missing-field-initializers -O2 -g -ggdb   -I../deps/hiredis -I../deps/linenoise -I../deps/lua/src >> .make-settings
echo PREV_FINAL_LDFLAGS=  -g -ggdb >> .make-settings
(cd ../deps && /Applications/Xcode.app/Contents/Developer/usr/bin/make hiredis linenoise lua)
(cd hiredis && /Applications/Xcode.app/Contents/Developer/usr/bin/make clean) > /dev/null || true
(cd linenoise && /Applications/Xcode.app/Contents/Developer/usr/bin/make clean) > /dev/null || true
(cd lua && /Applications/Xcode.app/Contents/Developer/usr/bin/make clean) > /dev/null || true
(cd jemalloc && [ -f Makefile ] && /Applications/Xcode.app/Contents/Developer/usr/bin/make distclean) > /dev/null || true
(rm -f .make-*)
(echo "" > .make-ldflags)
(echo "" > .make-cflags)
MAKE hiredis
cd hiredis && /Applications/Xcode.app/Contents/Developer/usr/bin/make static
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb  net.c
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb  hiredis.c
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb  sds.c
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb  async.c
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb  read.c
ar rcs libhiredis.a net.o hiredis.o sds.o async.o read.o
MAKE linenoise
cd linenoise && /Applications/Xcode.app/Contents/Developer/usr/bin/make
cc  -Wall -Os -g  -c linenoise.c
MAKE lua
cd lua/src && /Applications/Xcode.app/Contents/Developer/usr/bin/make all CFLAGS="-O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC='' " MYLDFLAGS="" AR="ar rcu"
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o lapi.o lapi.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o lcode.o lcode.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o ldebug.o ldebug.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o ldo.o ldo.c
ldo.c:496:7: warning: unused variable 'c' [-Wunused-variable]
  int c = luaZ_lookahead(p->z);
      ^
1 warning generated.
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o ldump.o ldump.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o lfunc.o lfunc.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o lgc.o lgc.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o llex.o llex.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o lmem.o lmem.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o lobject.o lobject.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o lopcodes.o lopcodes.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o lparser.o lparser.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o lstate.o lstate.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o lstring.o lstring.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o ltable.o ltable.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o ltm.o ltm.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o lundump.o lundump.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o lvm.o lvm.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o lzio.o lzio.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o strbuf.o strbuf.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o fpconv.o fpconv.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o lauxlib.o lauxlib.c
lauxlib.c:577:61: warning: while loop has empty body [-Wempty-body]
   while ((c = getc(lf.f)) != EOF && c != LUA_SIGNATURE[0]) ;
                                                            ^
lauxlib.c:577:61: note: put the semicolon on a separate line to silence this
      warning
1 warning generated.
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o lbaselib.o lbaselib.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o ldblib.o ldblib.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o liolib.o liolib.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o lmathlib.o lmathlib.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o loslib.o loslib.c
loslib.c:60:3: warning: 'tmpnam' is deprecated: This function is provided for
      compatibility reasons only. Due to security concerns inherent in the
      design of tmpnam(3), it is highly recommended that you use mkstemp(3)
      instead. [-Wdeprecated-declarations]
  lua_tmpnam(buff, err);
  ^
./luaconf.h:657:33: note: expanded from macro 'lua_tmpnam'
#define lua_tmpnam(b,e)         { e = (tmpnam(b) == NULL); }
                                       ^
/usr/include/stdio.h:277:7: note: 'tmpnam' has been explicitly marked deprecated
      here
char    *tmpnam(char *);
         ^
1 warning generated.
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o ltablib.o ltablib.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o lstrlib.o lstrlib.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o loadlib.o loadlib.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o linit.o linit.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o lua_cjson.o lua_cjson.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o lua_struct.o lua_struct.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o lua_cmsgpack.o lua_cmsgpack.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o lua_bit.o lua_bit.c
ar rcu liblua.a lapi.o lcode.o ldebug.o ldo.o ldump.o lfunc.o lgc.o llex.o lmem.o lobject.o lopcodes.o lparser.o lstate.o lstring.o ltable.o ltm.o lundump.o lvm.o lzio.o strbuf.o fpconv.o lauxlib.o lbaselib.o ldblib.o liolib.o lmathlib.o loslib.o ltablib.o lstrlib.o loadlib.o linit.o lua_cjson.o lua_struct.o lua_cmsgpack.o lua_bit.o	# DLL needs all object files
ranlib liblua.a
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o lua.o lua.c
cc -o lua  lua.o liblua.a -lm
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o luac.o luac.c
cc -O2 -Wall -DLUA_ANSI -DENABLE_CJSON_GLOBAL -DREDIS_STATIC=''    -c -o print.o print.c
cc -o luac  luac.o print.o liblua.a -lm
    CC adlist.o
    CC quicklist.o
    CC ae.o
    CC anet.o
    CC dict.o
    CC server.o
    CC sds.o
    CC zmalloc.o
    CC lzf_c.o
    CC lzf_d.o
    CC pqsort.o
    CC zipmap.o
    CC sha1.o
    CC ziplist.o
    CC release.o
    CC networking.o
    CC util.o
    CC object.o
    CC db.o
    CC replication.o
    CC rdb.o
    CC t_string.o
    CC t_list.o
    CC t_set.o
    CC t_zset.o
    CC t_hash.o
    CC config.o
    CC aof.o
    CC pubsub.o
    CC multi.o
    CC debug.o
    CC sort.o
    CC intset.o
    CC syncio.o
    CC cluster.o
    CC crc16.o
    CC endianconv.o
    CC slowlog.o
    CC scripting.o
    CC bio.o
    CC rio.o
    CC rand.o
    CC memtest.o
    CC crc64.o
    CC bitops.o
    CC sentinel.o
    CC notify.o
    CC setproctitle.o
    CC blocked.o
    CC hyperloglog.o
    CC latency.o
    CC sparkline.o
    CC redis-check-rdb.o
    CC geo.o
    CC lazyfree.o
    CC module.o
    CC evict.o
    CC expire.o
    CC geohash.o
    CC geohash_helper.o
    CC childinfo.o
    CC defrag.o
    CC siphash.o
    CC rax.o
    LINK redis-server
    INSTALL redis-sentinel
    CC redis-cli.o
    LINK redis-cli
    CC redis-benchmark.o
    LINK redis-benchmark
    INSTALL redis-check-rdb
    CC redis-check-aof.o
    LINK redis-check-aof

Hint: It's a good idea to run 'make test' ;)

/Users/lemire/CVS/github/redis-roaring
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb  net.c
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb  hiredis.c
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb  sds.c
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb  async.c
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb  read.c
cc -shared -Wl,-install_name,libhiredis.0.13.dylib -o libhiredis.dylib  net.o hiredis.o sds.o async.o read.o
ar rcs libhiredis.a net.o hiredis.o sds.o async.o read.o
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb  test.c
cc -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb  -o hiredis-test   test.o libhiredis.a
Generating hiredis.pc for pkgconfig...
/Users/lemire/CVS/github/redis-roaring
```

I also find it suspicious, but with very little experience with compiling C on OSX, it's not immediately meaningful to me.

```$ cc --version
Apple LLVM version 8.1.0 (clang-802.0.42)
Target: x86_64-apple-darwin16.7.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

```$ make clean
rm -rf libhiredis.dylib libhiredis.a hiredis-test hiredis.pc examples/hiredis-example* *.o *.gcda *.gcno *.gcov```

```$ make
cc -std=c99 -pedantic -c -O3 -fPIC  -Wall -W -Wstrict-prototypes -Wwrite-strings -g -ggdb x86_64 net.c
clang: error: no such file or directory: 'x86_64'
make: *** [net.o] Error 1```

```$ ./hello
hello```

`declare -x ARCH="x86_64"`, set by an overzealous customization script. That's one mystery solved (thanks a bunch - I probably never would have figured that out myself), now `./configure.sh` completes successfully, and then `make` fails:

```
/Users/abernstein/src/redis-roaring/src/type.c:44:7: error: use of GNU old-style field designator extension
      [-Werror,-Wgnu-designator]
      aof: aof,
      ^~~~
      .aof =
/Users/abernstein/src/redis-roaring/src/type.c:45:7: error: use of GNU old-style field designator extension
      [-Werror,-Wgnu-designator]
      key: key
      ^~~~
      .key =
2 errors generated.
```

After changing lines 43-46 in type.c from

```
  Bitmap_aof_rewrite_callback_params params = {
      aof: aof,
      key: key
  };
```

to

```
  Bitmap_aof_rewrite_callback_params params = {
      .aof = aof,
      .key= key
  };
```
`make` appears to complete successfully, which seems odd.

--- a/src/type.c
+++ b/src/type.c
@@ -41,8 +41,8 @@ static bool BitmapAofRewriteCallback(uint32_t offset, void* param) {
 void BitmapAofRewrite(RedisModuleIO* aof, RedisModuleString* key, void* value) {
   Bitmap* bitmap = value;
   Bitmap_aof_rewrite_callback_params params = {
-      aof: aof,
-      key: key
+      .aof: aof,
+      .key: key
   };
   roaring_iterate(bitmap, BitmapAofRewriteCallback, &params);
 }

--- a/src/type.c
+++ b/src/type.c
@@ -41,8 +41,8 @@ static bool BitmapAofRewriteCallback(uint32_t offset, void* param) {
 void BitmapAofRewrite(RedisModuleIO* aof, RedisModuleString* key, void* value) {
   Bitmap* bitmap = value;
   Bitmap_aof_rewrite_callback_params params = {
-      .aof: aof,
-      .key: key
+      .aof = aof,
+      .key = key
   };
   roaring_iterate(bitmap, BitmapAofRewriteCallback, &params);
 }

